### PR TITLE
S3 Improvements for test connection and connector status

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

- Handle test connection for S3
- Add new status handling for the S3 ingestion
- Update the endpoint string to `objectStoreService` to ease an issue where the Ingestion Pipeline service type was sent wrongfully as `objectstoreService` instead of `objectStoreService`. This is ok for now as a followup, we'll do a refactor from `objectStoreService` to the final `StorageService`.
- Update list field param name
- Fix testing the connection and deploying object store services from the UI

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/objectstore/objectstore_service.py [hunk 1]
```diff
@@ -75,27 +75,12 @@ class ObjectStoreServiceTopology(ServiceTopology):
     )
 
 
-class ObjectStoreSourceStatus(SourceStatus):
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/objectstore/objectstore_service.py:78]: Using the new status inherited from `Source`

File: openmetadata-service/src/main/java/org/openmetadata/service/secrets/converter/ClassConverterFactory.java [hunk 2]
```diff
@@ -42,16 +42,16 @@ public class ClassConverterFactory {
             new HashMap<>() {
               {
                 put(AirflowConnection.class, new AirflowConnectionClassConverter());
+                put(BigQueryConnection.class, new BigQueryConnectionClassConverter());
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/secrets/converter/ClassConverterFactory.java:45]: @nahuelverdugo im sorting this alphabetically

File: openmetadata-spec/src/main/resources/json/schema/entity/services/objectstoreService.json [hunk 3]
```diff
@@ -46,12 +38,6 @@
           "oneOf": [
             {
               "$ref": "connections/objectstore/s3ObjectStoreConnection.json"
-            },
-            {
-              "$ref": "connections/objectstore/gcsObjectStoreConnection.json"
-            },
-            {
-              "$ref": "connections/objectstore/azureObjectStoreConnection.json"
```
Comment (pmbrull) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/objectstoreService.json:54]: Removing this from the JSON Schema for now. There is no implementation on the ingestion side yet

File: openmetadata-ui/src/main/resources/ui/src/utils/ServiceUtils.tsx [hunk 4]
```diff
@@ -613,17 +605,17 @@ export const shouldTestConnection = (serviceType: string) => {
...
+    [ServiceCategory.DASHBOARD_SERVICES]: ServiceType.Dashboard,
+    [ServiceCategory.DATABASE_SERVICES]: ServiceType.Database,
+    [ServiceCategory.MESSAGING_SERVICES]: ServiceType.Messaging,
+    [ServiceCategory.ML_MODEL_SERVICES]: ServiceType.MlModel,
+    [ServiceCategory.METADATA_SERVICES]: ServiceType.Metadata,
+    [ServiceCategory.OBJECT_STORE_SERVICES]: ServiceType.ObjectStore,
+    [ServiceCategory.PIPELINE_SERVICES]: ServiceType.Pipeline,
+  };
+
+  return serviceTypeMap[serviceCat];
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/ServiceUtils.tsx:618]: Would be great to have it in const 
cc: @Sachin-chaurasiya

**Review by chirag-madlani — APPROVED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/services/objectstore/ObjectStoreServiceResource.java [hunk 5]
```diff
@@ -58,15 +58,15 @@
...
+@Path("/v1/services/objectStoreServices")
 @Api(value = "Object store service collection", tags = "Services -> Object store service collection")
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
 @Collection(name = "objectstoreServices")
 public class ObjectStoreServiceResource
     extends ServiceEntityResource<ObjectStoreService, ObjectStoreServiceRepository, ObjectStoreConnection> {
 
-  public static final String COLLECTION_PATH = "v1/services/objectstoreServices/";
+  public static final String COLLECTION_PATH = "v1/services/objectStoreServices/";
```
[the remark this answers is not in the document]
Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/services/objectstore/ObjectStoreServiceResource.java:69]: yeah, I see what you mean. Better to stick to the same naming IMO. We could discuss this offline if this is something we want to land for 1.0

**Pull request merged**